fs-cache/src/memory_limited_storage.rs

```diff
@@ -0,0 +1,273 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+use data_error::{ArklibError, Result};
+use linked_hash_map::LinkedHashMap;
+
+pub struct MemoryLimitedStorage<K, V> {
+    /// Label for logging
+    label: String,
+    /// Path to the underlying folder where data is persisted
+    path: PathBuf,
```

Comment:
Why do we need a separate folder for persistence (where each key-value pair stored as a different file)? Can't we just use a single file? 

If each value can be serialized to a string, we could store all values in a single JSON file.

Reply:
@tareknaser `FolderStorage` is primary storage for cache, because values can be large, e.g. previews.

With time, it would be great to also implement cache backed by `FileStorage`. And sometime we're going to implement `ChunkedStorage` as well (hybrid between the two flavours).

Reply:
> because values can be large, e.g. previews

Right, that makes sense.